transform/enrich_kv_store_set_add.go

```diff
@@ -0,0 +1,180 @@
+package transform
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/brexhq/substation/config"
+	iconfig "github.com/brexhq/substation/internal/config"
+	"github.com/brexhq/substation/internal/errors"
+	"github.com/brexhq/substation/internal/kv"
+	"github.com/brexhq/substation/message"
+)
+
+type enrichKVStoreSetAddObjectConfig struct {
+	// TTLKey retrieves a value from an object that is used as the time-to-live (TTL)
+	// of the item set into the KV store. This value must be an integer that represents
+	// the Unix time when the item will be evicted from the store. Any precision greater
+	// than seconds (e.g., milliseconds, nanoseconds) is truncated to seconds.
+	//
+	// This is optional and defaults to using no TTL when setting items into the store.
+	TTLKey string `json:"ttl_key"`
+
+	iconfig.Object
+}
+
+type enrichKVStoreSetAddConfig struct {
+	// Prefix is prepended to the key and can be used to simplify
+	// data management within a KV store.
+	//
+	// This is optional and defaults to an empty string.
+	Prefix string `json:"prefix"`
+	// TTLOffset is an offset used to determine the time-to-live (TTL) of the item set
+	// into the KV store. If Object.TTLKey is configured, then this value is added to the TTL
+	// value retrieved from the object. If Object.TTLKey is not used, then this value is added
+	// to the current time.
+	//
+	// For example, if Object.TTLKey is not set and the offset is "1d", then the value
+	// will be evicted from the store when more than 1 day has passed.
+	//
+	// This is optional and defaults to using no TTL when setting values into the store.
+	TTLOffset string `json:"ttl_offset"`
+	// CloseKVStore determines if the KV store is closed when a control
+	// message is received.
+	//
+	// This is optional and defaults to false (KV store is not closed).
+	CloseKVStore bool `json:"close_kv_store"`
+
+	ID      string                          `json:"id"`
+	Object  enrichKVStoreSetAddObjectConfig `json:"object"`
+	KVStore config.Config                   `json:"kv_store"`
+}
+
+func (c *enrichKVStoreSetAddConfig) Decode(in interface{}) error {
+	return iconfig.Decode(in, c)
+}
+
+func (c *enrichKVStoreSetAddConfig) Validate() error {
+	if c.Object.SourceKey == "" {
+		return fmt.Errorf("object_source_key: %v", errors.ErrMissingRequiredOption)
+	}
+
+	if c.Object.TargetKey == "" {
+		return fmt.Errorf("object_target_key: %v", errors.ErrMissingRequiredOption)
+	}
+
+	if c.KVStore.Type == "" {
+		return fmt.Errorf("kv_store: %v", errors.ErrMissingRequiredOption)
+	}
+
+	return nil
+}
+
+func newenrichKVStoreSetAdd(_ context.Context, cfg config.Config) (*enrichKVStoreSetAdd, error) {
```

Comment:
Style: Should this name be cased in [MixedCaps](https://go.dev/doc/effective_go#mixed-caps)?
```suggestion
func newEnrichKVStoreSetAdd(_ context.Context, cfg config.Config) (*enrichKVStoreSetAdd, error) {
```